### ISSUE TITLE
Manual diff fragmentation

### ISSUE THREAD

**Issue by Shatur** [lifescapegame/bevy_replicon#16]
Currently we sending one message per diff and rely on the Renet fragmentation. If the message is large, then Renet divides it into several parts.
But if packet loss is present, then the loss of even one part results in the loss of the entire diff.
To avoid this we should send diff per-entity. To achieve this we need to track last acknowledged tick per entity.

**Comment by RJ** [comment]
if the server has sent, and the client has acked, replication data for tick A, and the server is now on tick C and scheduled to send out replication data - do we really need to send tick-B data to the client too? 

i suppose that's game-specific, but i certainly don't care about receiving tick B & C together in one packet  - i'd discard the B state and apply C if i received them both together.

**Comment by Shatur** [comment]
Yes, that's the problem. Sending only last tick value for each component is important optimization.
I currently thinking that current approach + using rooms for packet fragmentation is the only way to go.

**Comment by Shatur** [comment]
I looked at the analysis of the [Quake 3 network code](https://fabiensanglard.net/quake3/network.php) and looked into Unreal Engine (source code on GitHub is available after request) and they use the approach I described - send the last value unacknowledged value all the time.
So I don't think that using per-tick will help with packet fragmentation. If we decide to separate sending data per tick, if we include only the last changed value in the last tick, we will need to include it into a single message to avoid partial replication :(

I would suggest manual approach with rooms for now.
